### PR TITLE
Improved the hud

### LINKED ISSUE #468: HUD update
As a player I want have a better HUD, so that I can see my status more easily.

This task has the following subtasks:
- #469 HUD update - Holdable turning
- #470 HUD update - Health display
- #471 HUD update - Game ending indication
- #472 HUD update - Bomb timer
- #473 HUD update - HUD element positions

### LINKED ISSUE #469: HUD update - Holdable turning
Make the holdable follow the player correctly in VR mode. Currently, held items are not held in the direction that the player is facing.

This is a subtask of #468 - HUD update

### LINKED ISSUE #470: HUD update - Health display
Fix rounding of health and show half hearts.

This is a subtask of #468 - HUD update

### LINKED ISSUE #471: HUD update - Game ending indication
Show the player a short screen indicating that the game has finished

This is a subtask of #468 - HUD update

### LINKED ISSUE #472: HUD update - Bomb timer
Remove the bomb countdown timer after the bomb has exploded.

This is a subtask of #468 - HUD update

### LINKED ISSUE #473: HUD update - HUD element positions
Move all HUD elements to a place that is visible for the player

This is a subtask of #468 - HUD update

### PR DESCRIPTION
**Pull request by ChielBruin**
- Fixed heart rounding issues
- Added half heart display
- Improved game timer position
- Added popup texts on game state switches
- Fixed holdable orientations in VR
- Fixed the bomb timer

Closes #468, closes #469, closes #470, closes #471, closes #472 and closes #473

### CODE REVIEW

**Comment by Taeir** [comment]
Wow, many pretty good small changes here, good job.

The pull request looks quite big, but most of the changes are from code that has been moved or from classes that have been renamed.
The refactoring of GameController was long overdue it seems, the code has been cleaned up quite significantly.

I think this can be merged.

**Comment by WytzeElhorst** [comment]
Seem like generally good improvements to both the hud and the framework, can be merged

File: src/main/java/nl/tudelft/contextproject/controller/EndingController.java [hunk 1]
```diff
@@ -25,6 +27,16 @@ public EndingController(Application app, boolean elvesWin) {
 	}
 	
 	@Override
+	public void initialize(AppStateManager stateManager, Application app) {
+		super.initialize(stateManager, app);
+		if (elvesWin) {
+			getHUD().showPopupText("You've won!", ColorRGBA.Green, 4);
+		} else {
+			getHUD().showPopupText("You've lost!", ColorRGBA.Red, 4);
```
Comment (Diocruel) [inline src/main/java/nl/tudelft/contextproject/controller/EndingController.java:35]: Perhaps You've died! or perished? is better here.

File: src/main/java/nl/tudelft/contextproject/controller/EndingController.java [hunk 2]
```diff
@@ -25,6 +27,16 @@ public EndingController(Application app, boolean elvesWin) {
 	}
 	
 	@Override
+	public void initialize(AppStateManager stateManager, Application app) {
+		super.initialize(stateManager, app);
+		if (elvesWin) {
+			getHUD().showPopupText("You've won!", ColorRGBA.Green, 4);
```
Comment (Diocruel) [inline src/main/java/nl/tudelft/contextproject/controller/EndingController.java:33]: Perhaps "You have found the treasure and saved the kingdom from bankruptcy!" is better here?

File: src/main/java/nl/tudelft/contextproject/controller/GameThreadController.java [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+	public static final float MAX_TPF = .033f;
+
+	private Game game;
+	private HUD hud;
+
+	/**
+	 * Constructor for a game thread.
+	 *
+	 * @param app
+	 * 		The Main instance of this game
```
Comment (Diocruel) [inline src/main/java/nl/tudelft/contextproject/controller/GameThreadController.java:47]: remove capital T

File: src/main/java/nl/tudelft/contextproject/controller/GameThreadController.java [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+	 * @param level
+	 * 		The level for this game
```
Comment (Diocruel) [inline src/main/java/nl/tudelft/contextproject/controller/GameThreadController.java:49]: remove capital T

**Pull request merged**